ethers-solc/src/config.rs

```diff
@@ -146,6 +146,67 @@ impl ProjectPathsConfig {
     pub fn find_libs(root: impl AsRef<Path>) -> Vec<PathBuf> {
         vec![utils::find_fave_or_alt_path(root, "lib", "node_modules")]
     }
+
+    /// Flatten all file imports into a single string
+    pub fn flatten(&self, target: &Path) -> Result<String> {
+        tracing::trace!("flattening file");
+        let graph = Graph::resolve(self)?;
+
+        struct Flattener<'a> {
+            f: &'a dyn Fn(&Flattener, &Path) -> Result<String>,
+        }
+        let flatten = Flattener {
+            f: &|flattener, target| {
+                let target_dir = target.parent().ok_or_else(|| {
+                    SolcError::msg(format!(
+                        "failed to get parent directory for \"{:?}\"",
+                        target.display()
+                    ))
+                })?;
+                let target_index = graph.files().get(target).ok_or_else(|| {
+                    SolcError::msg(format!("cannot resolve file at \"{:?}\"", target.display()))
+                })?;
+                let target_node = graph.node(*target_index);
+
+                let mut imports = target_node.imports().clone();
+                imports.sort_by(|a, b| b.loc().0.cmp(&a.loc().0));
+
+                let content = target_node.content().bytes().collect::<Vec<_>>();
+                let mut curr_import = imports.pop();
+                let mut extended = vec![];
+
+                let mut i = 0;
+                while i < content.len() {
```

Comment:
there should also be `[u8].advance` to shift,
seems like that's what's going on in this loop?